File: Meraki_practice/dutch_flag.rb
#!/usr/bin/ruby
###############################################################################
# Name        : dutch_flag.rb                                                 #
# Author      : Abel Gancsos                                                  #
# Version     : v. 1.0.0.0                                                    #
# Description : Practice working with Dutch Flag problem.                     #
###############################################################################

class DutchFlag
	@numbers=nil;
	def initialize(params={})
		@numbers = params["--list"] == nil ? [1,2,3,4,5] : params["--list"];
	end
	def invoke()
		print "Original: #{@numbers}\n";
		sorted = @numbers.sort_by {|v| (v % 2 != 0) ? 0 : 1};
		print "Sorted  : #{sorted}\n";
	end
end

params = {};
for i in 0..ARGV.length - 1
	params[ARGV[i]] = ARGV[i + 1]
end

session = DutchFlag.new(params);
session.invoke();

File: Meraki_practice/dutch_flag.py
#!/usr/bin/env python3
###############################################################################
# Name        : dutch_flag.py                                                 #
# Author      : Abel Gancsos                                                  #
# Version     : v. 1.0.0.0                                                    #
# Description : Practice working with Dutch Flag problem.                     #
###############################################################################
import os, sys;

class DutchFlag:
	numbers=None;
	def __init__(self, params=dict()):
		self.numbers = params["--list"].split(",") if "--list" in params.keys() else [1,2,3,4,5];
		pass;
	def invoke(self):
		print("Original: {0}".format(self.numbers));
		sorted_value = sorted(self.numbers, key=lambda x:x % 2 == 0);
		print("Sorted  : {0}".format(sorted_value));
	pass;

if __name__ == "__main__":
	params = dict();
	for i in range(0, len(sys.argv) - 1): params[sys.argv[i]] = sys.argv[i + 1];
	session = DutchFlag(params);
	session.invoke();
	pass;
